caffe2/core/common_gpu.h

```diff
@@ -26,7 +26,7 @@
 #include <c10/cuda/CUDAGuard.h>
 
 #define CAFFE2_CUDA_EXPORT C10_EXPORT
-#define CAFFE2_CUDA_API TORCH_CUDA_API
+#define CAFFE2_CUDA_API TORCH_CUDA_CPP_API
```

Comment:
I forget; why does this line only define `TORCH_CUDA_CPP_API` rather than also defining `TORCH_CUDA_CU_API` (as is done in `c10/macros/Export.h`)?

Reply:
This is because the declarations that have this macro will probably be in the `torch_cuda_cpp` target and not `torch_cuda_cu`.

And yes, thank you for the summary. The code in `tools/shared/cwrap_common.py` just changes how the THNN code gets parsed (switching TORCH_CUDA_API to either TORCH_CUDA_CU_API or TORCH_CUDA_CPP_API).